instrumentation/rxjava-2.0/library/src/main/java/io/opentelemetry/instrumentation/rxjava2/TracingAssembly.java

```diff
@@ -0,0 +1,318 @@
+      oldOnObservableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super ConnectableObservable, ? extends ConnectableObservable>
+      oldOnConnectableObservableAssembly;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<
+          ? super Completable, ? super CompletableObserver, ? extends CompletableObserver>
+      oldOnCompletableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Single, ? super SingleObserver, ? extends SingleObserver>
+      oldOnSingleSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Maybe, ? super MaybeObserver, ? extends MaybeObserver>
+      oldOnMaybeSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Flowable, ? super Subscriber, ? extends Subscriber>
+      oldOnFlowableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super ConnectableFlowable, ? extends ConnectableFlowable>
+      oldOnConnectableFlowableAssembly;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super ParallelFlowable, ? extends ParallelFlowable>
+      oldOnParallelAssembly;
+
+  @GuardedBy("TracingAssembly.class")
+  private static boolean enabled;
+
+  private TracingAssembly() {}
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  public static synchronized void enable() {
+    if (enabled) {
+      return;
+    }
+
+    enableObservable();
+
+    oldOnConnectableObservableAssembly = RxJavaPlugins.getOnConnectableObservableAssembly();
+    RxJavaPlugins.setOnConnectableObservableAssembly(
+        compose(
+            oldOnConnectableObservableAssembly,
+            new ConditionalOnCurrentContextFunction<ConnectableObservable>() {
+              @Override
+              ConnectableObservable applyActual(
+                  ConnectableObservable connectableObservable, Context ctx) {
+                return new TracingConnectableObservable(connectableObservable, ctx);
+              }
+            }));
+
+    enableCompletable();
+
+    enableSingle();
+
+    enableMaybe();
+
+    enableFlowable();
+
+    oldOnConnectableFlowableAssembly = RxJavaPlugins.getOnConnectableFlowableAssembly();
+    RxJavaPlugins.setOnConnectableFlowableAssembly(
+        compose(
+            oldOnConnectableFlowableAssembly,
+            new ConditionalOnCurrentContextFunction<ConnectableFlowable>() {
+              @Override
+              ConnectableFlowable applyActual(
+                  ConnectableFlowable connectableFlowable, Context ctx) {
+                return new TracingConnectableFlowable(connectableFlowable, ctx);
+              }
+            }));
```

Comment:
Since you've extracted `enableSingle()`, `enableMaybe()`, ... private method can you do it for all functions instrumented here?

Reply:
👍